chain/chain/src/resharding.rs

```diff
@@ -39,12 +38,27 @@ pub struct StateSplitResponse {
     pub new_state_roots: Result<HashMap<ShardUId, StateRoot>, Error>,
 }
 
+fn get_trie_update_batch(
+    iter: &mut impl Iterator<Item = (Vec<u8>, Vec<u8>)>,
+) -> Option<Vec<(Vec<u8>, Option<Vec<u8>>)>> {
+    let mut size: u64 = 0;
+    let mut entries = Vec::new();
+    while let Some((key, value)) = iter.next() {
+        size += value.len() as u64;
+        entries.push((key, Some(value)));
+        if size > STATE_PART_MEMORY_LIMIT.as_u64() {
+            break;
+        }
+    }
+    (!entries.is_empty()).then_some(entries)
```

Comment:
1. What does then_some do? (imo this is an obscure function and I'd vote for an if expression instead for readability)
2. Why are we returning either None or a Some(non empty vector)? What if we just return a vector that may be empty?

Reply:
I was looking for a good one liner to convert entries to Some(entries) based on whether it's empty or not, and the best I could come up with was the `then_some` function. Anyway, will change it to explicit if else 

We are using the `while let Some(batch) = ... {}` patter here which has better type guarantees.